src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -124,6 +127,35 @@ protected function json($data, $status = 200, $headers = array(), $context = arr
     }
 
     /**
+     * Returns a BinaryFileResponse object with original or customized file name and disposition header.
+     *
+     * @param File|string $file        File object or path to file to be sent as response
+     * @param string|null $fileName    File name to be sent to response or null (will use original file name)
+     * @param string      $disposition Disposition of response ("attachment" is default, other type is "inline")
+     *
+     * @return BinaryFileResponse
+     */
+    protected function file($file, $fileName = null, $disposition = ResponseHeaderBag::DISPOSITION_ATTACHMENT)
+    {
+        if (!is_string($file) && !$file instanceof File) {
+            throw new \InvalidArgumentException(sprint('The "%s" method expects first argument to be a string or an instance of "%s"', __METHOD__, File::class));
+        }
+
+        if (!$file instanceof File) {
+            $file = new File($file);
+        }
+
+        $response = new BinaryFileResponse($file);
+        $mimeType = $file->getMimeType();
+
+        $response->headers->set('Content-Type', $mimeType);
+        $disposition = $response->headers->makeDisposition($disposition, $fileName === null ? $file->getFileName() : $fileName);
+        $response->headers->set('Content-Disposition', $disposition);
```

Comment:
It would be better to use the `setContentDisposition()` of `BinaryFileResponse`:

```php
$response->setContentDisposition($disposition, $fileName === null ? $file->getFileName() : $fileName);
```